recipes/log.c/all/conanfile.py

```diff
@@ -0,0 +1,63 @@
+from conans import ConanFile, CMake, tools
+import glob
+from conans.errors import InvalidNameException
+
+class logcConan(ConanFile):
+    name = "log.c"
+    version = "0.1.0"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/rxi/log.c"
+    description = "A simple logging library implemented in C99"
+    topics = ("logging", "log", "logging-library", "conan", "logc","purec","c99")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "log_use_color": [True, False] }
+    default_options = { "shared": False,
+                        "fPIC": True,
+                        "log_use_color": True }
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+       
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        name = glob.glob("log.c-*")
+        print(name)
+        if len(name) != 1:
+          raise InvalidNameException("Invalid zip with source name")
+        tools.rename(name[0],self._source_subfolder)
```

Comment:
Just rename.
An `IndexError` will be thrown when nothing is found.
The packager should check whether the archive contains multiple folders.
The sha256 in `conandata.yml` will prevent suddenly having multiple subdirectories.

```suggestion
        tools.rename(glob.glob("log.c-*")[0],self._source_subfolder)
```